Funcky.SourceGenerator/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,28 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Funcky.SourceGenerator;
+
+internal static class SyntaxNodeHelper
+{
+    public static TSyntaxNode? TryGetParentSyntax<TSyntaxNode>(this SyntaxNode syntaxNode)
+        where TSyntaxNode : SyntaxNode
+        => syntaxNode.Parent switch
+        {
+            null => null,
+            TSyntaxNode parent => parent,
+            _ => syntaxNode.Parent.TryGetParentSyntax<TSyntaxNode>(),
```

Comment:
Just an idea, not a strong preference
```suggestion
            var parent => parent.TryGetParentSyntax<TSyntaxNode>(),
```